p2p/peer.py

```diff
@@ -497,6 +545,30 @@ class LESPeer(BasePeer):
         # TODO: Disconnect if the remote doesn't serve headers.
         self.head_info = cmd.as_head_info(msg)
 
+    async def _get_headers_at_chain_split(
+            self, block_number: BlockNumber) -> Tuple[BlockHeader, BlockHeader]:
+        request_id = gen_request_id()
+        self.sub_proto.send_get_block_headers(
+            block_number - 1, max_headers=2, request_id=request_id, reverse=False)
+        while True:
+            cmd, msg = await self.read_msg()
+            msg = cast(Dict[str, Any], msg)
+            if not isinstance(cmd, les.BlockHeaders):
+                self.logger.debug(
+                    "Ignoring %s msg from %s as we haven't performed the chain-split check yet",
+                    cmd, self)
+                continue
+            elif msg['request_id'] != request_id:
+                self.logger.debug(
+                    "Got a BlockHeaders msg with the wrong request_id (%d) from %s, ignoring",
+                    msg['request_id'], self)
+                continue
+            headers = msg['headers']
+            if len(headers) != 2:
+                raise HeaderNotFound()
+            parent, header = headers
```

Comment:
We could perform all those extra checks you describe, but it'd make the code a lot more complex for no real benefit because a malicious peer that passes this check by providing fake data would be caught later on when we actually try to import headers. That's what I had in mind when I wrote the comment in the except block that handles `HeaderNotFound`.

Unless I'm missing something, I don't think this could be an attack vector so I'd rather avoid the extra complexity

Reply:
I think the added complexity could be as simple as a single function call to a validation utility.

```
def validate_header_response(request_params, response_headers):
    ...
```

Should be small and easy to insert into any function body which requests and receives headers (though it does get complicated if we have call sites which don't have access to the `request_params`).

**IF** it is as simple as this then I feel strongly enough to push for it.  If there are hidden complexities that I don't see, then I'll happily defer to your judgement.

My reasoning for still wanting this even with the knowledge that the import validation **should** catch things is that unless I **know with near certaintly** that something isn't an attack vector, I try to assume that something is hiding in there, and skipping validation seems like a good avenue for security holes to find their way in.

Reply:
Indeed, that's straightforward enough. Pushed a new commit with the validation function you suggested